internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -7,26 +7,152 @@ import (
 	"testing"
 	"time"
 
-	"github.com/go-logr/logr"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime/runtimefakes"
 	ngxclient "github.com/nginxinc/nginx-plus-go-client/client"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
-var _ = Describe("NGINX Runtime Manager", func() {
+var _ = Describe("NGINX Runtime Manager", Ordered, func() {
 	It("returns whether or not we're using NGINX Plus", func() {
-		mgr := NewManagerImpl(nil, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr := runtime.NewManagerImpl(nil, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeFalse())
 
-		mgr = NewManagerImpl(&ngxclient.NginxClient{}, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr = runtime.NewManagerImpl(&ngxclient.NginxClient{}, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeTrue())
 	})
+
+	var (
+		manager         runtime.Manager
+		upstreamServers []ngxclient.UpstreamServer
+		upstreamServer  ngxclient.UpstreamServer
+		ngxPlusClient   *runtimefakes.FakeNginxPlusClient
+		process         *runtimefakes.FakeProcessHandler
+
+		metrics      *runtimefakes.FakeMetricsCollector
+		verifyClient *runtimefakes.FakeVerifyClient
+	)
+
+	var (
+		MaxConnsMock = 150
+		MaxFailsMock = 20
+		WeightMock   = 10
+		BackupMock   = false
+		DownMock     = false
+	)
+
+	BeforeAll(func() {
+		upstreamServer = ngxclient.UpstreamServer{
+			ID:          1,
+			Server:      "10.0.0.1:80",
+			MaxConns:    &MaxConnsMock,
+			MaxFails:    &MaxFailsMock,
+			FailTimeout: "test",
+			SlowStart:   "test",
+			Route:       "test",
+			Backup:      &BackupMock,
+			Down:        &DownMock,
+			Drain:       false,
+			Weight:      &WeightMock,
+			Service:     "test",
+		}
+
+		upstreamServers = []ngxclient.UpstreamServer{
+			upstreamServer,
+		}
+
+	})
+
+	Context("Reload", func() {
+		BeforeAll(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			process = &runtimefakes.FakeProcessHandler{}
+			metrics = &runtimefakes.FakeMetricsCollector{}
+			verifyClient = &runtimefakes.FakeVerifyClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+		})
+
+		It("sucessfully reloads NGINX configuration", func() {
+			err := manager.Reload(context.Background(), 1)
+			Expect(err).To(BeNil())
```

Comment:
```suggestion
			Expect(manager.Reload(context.Background(), 1)).To(Succeed())	
```
I would also verify that each of the other interface functions were called (WaitForCorrectVersionCallCount, SyscallKillCallCount, etc)